pkg/bloomgateway/util.go

```diff
@@ -184,3 +184,29 @@ func partitionRequest(req *logproto.FilterChunkRefRequest) []seriesWithBounds {
 
 	return result
 }
+
+type MultiFingerprintBounds []v1.FingerprintBounds
+
+func (mb MultiFingerprintBounds) Union(target v1.FingerprintBounds) MultiFingerprintBounds {
```

Comment:
I think for this to behave as expected, the multi-bounds must be
a) sorted already
b) contain disjoint (non-overlapping) ranges

Otherwise you can end up with many overlapping ranges in the result set as well.

Consider:
```
multi = [(0,6), (5,15)]
target = (8,10)

result = [(0,6), (5,15)] // can be further reduced
```

Thinking about it, it's probably not a real concern we have if we can ensure the multi-bounds are ordered and non overlapping.

Reply:
I solved the issue by using a different algorithm.